packages/core/src/Drawer/Drawer.tsx

```diff
@@ -67,108 +56,135 @@ export const Drawer: React.FC<DrawerProps> = ({
   onClose,
   onCollapse,
   placement = 'right',
+
+  // neighborhood-related props:
+  snapHeights,
+  snapDimensions,
+  disableEnterAnimation,
   ...props
 }) => {
   const { boxShadow, onScrollHandler } = useScrollWithShadow()
+  const { snapPosition, handleSnap } = useSnap(snapHeights)
+  const SnapContainer = snapHeights ? motion.div : React.Fragment
+
   return (
-    <AnimatePresence>
-      {isOpen && (
-        <DrawerWrapper
-          placement={isMobile ? 'bottom' : placement}
-          padding={isFloating ? 3 : 0}
-          maxHeight={isMobile ? ['290px', '400px', '480px', 'calc(100vh - 64px)'] : props.height ?? '100%'}
-          maxWidth={isMobile ? '100%' : ['400px', '600px', '800px', '100%']}
-          width={isMobile ? '100%' : props.width}
-          {...props}
-          height={!isCollapsed && props.height ? props.height : 'fit-content'}
-        >
-          <DrawerRoot
-            data-testid='drawer'
-            isCollapsed={isCollapsed}
-            isFloating={isFloating}
-            key='drawer'
+    <SnapContainer
```

Comment:
It won't affect the existing behavior no, and ideally it shouldn't render (which is why on line 68 I made a toggle) - open to suggestions on other approaches (e.g. is it better to have a null rather than react.fragement?)